File: Java/selectionSort.java
public class SelectionSort {

    // Method to perform Selection Sort
    public static void selectionSort(int[] array) {
        int n = array.length; // Get the length of the array

        // Traverse through all elements in the array
        for (int i = 0; i < n - 1; i++) {
            // Assume the first element of the unsorted part is the minimum
            int minIndex = i;

            // Check the rest of the array for a smaller element
            for (int j = i + 1; j < n; j++) {
                // Update minIndex if a smaller element is found
                if (array[j] < array[minIndex]) {
                    minIndex = j; // Update the index of the minimum element
                }
            }

            // Swap the found minimum element with the first element of the unsorted part
            if (minIndex != i) {
                int temp = array[i];
                array[i] = array[minIndex];
                array[minIndex] = temp;
            }
        }
    }

    // Method to print the array
    public static void printArray(int[] array) {
        // Iterate through the array and print each element
        for (int value : array) {
            System.out.print(value + " ");
        }
        System.out.println();
    }

    // Main method to test the sorting algorithm
    public static void main(String[] args) {
        // Sample array to be sorted
        int[] array = {64, 34, 25, 12, 22, 11, 90};

        System.out.println("Original array:");
        printArray(array); // Print the original array

        // Call the selection sort method
        selectionSort(array);

        System.out.println("Sorted array:");
        printArray(array); // Print the sorted array
    }
}

File: C++/selectionSort.cpp
#include <iostream>
#include <vector>

// Function to implement Selection Sort
void selectionSort(std::vector<int>& arr) {
    int n = arr.size(); // Get the size of the array

    // Outer loop to iterate over the unsorted portion of the array
    for (int i = 0; i < n - 1; ++i) {
        int minIndex = i; // Assume the first element is the minimum

        // Inner loop to find the minimum element in the unsorted portion of the array
        for (int j = i + 1; j < n; ++j) {
            // If a smaller element is found, update minIndex
            if (arr[j] < arr[minIndex]) {
                minIndex = j;
            }
        }

        // Swap the minimum element found with the first element of the unsorted portion
        if (minIndex != i) {
            std::swap(arr[i], arr[minIndex]);
        }
    }
}

// Helper function to print the elements of the array
void printArray(const std::vector<int>& arr) {
    for (int i = 0; i < arr.size(); ++i) {
        std::cout << arr[i] << " "; // Print each element followed by a space
    }
    std::cout << std::endl; // Print a newline at the end
}

// Main function
int main() {
    // Example array to sort
    std::vector<int> arr = {64, 25, 12, 22, 11};

    // Output the original array
    std::cout << "Original array: ";
    printArray(arr);

    // Call the selection sort function
    selectionSort(arr);

    // Output the sorted array
    std::cout << "Sorted array: ";
    printArray(arr);

    return 0;
}
